### PR TITLE
Schedules

### PR DESCRIPTION
**Pull request by SayantanKhanra10**
Adding support for creating Schedules in a Job Template.

### CODE REVIEW

File: towerlib/entities/host.py [hunk 1]
```diff
@@ -283,24 +283,24 @@ def recent_jobs(self):
...
             api=self._tower.api, id=self.id)
         response = self._tower.session.get(url)
-        if response.ok:
-            return response.json
-        else:
+        if not response.ok:
             self._logger.error('Error finding ansible facts for {name}.'.format(
                 name=self._data.get('name')))
-            return '{}'
+        return response.json if response.ok else '{}'
```
Comment (costastf) [inline towerlib/entities/host.py:302]: response.json is a method. you probably want to return response.json()

File: towerlib/entities/schedule.py [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+
+
+class Schedule(Entity):
+    """Models the schedule entity of ansible tower."""
+
+    def __init__(self, tower_instance, data):
+        Entity.__init__(self, tower_instance, data)
+
+    @property
+    def rrule(self):
```
Comment (costastf) [inline towerlib/entities/schedule.py:47]: wouldn't it be friendlier to actually call the variable "recurrence_rule"?

File: towerlib/entities/schedule.py [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+        """Enables processing of this schedule.
+
+        Returns:
+            bool: Enables processing of this schedule.
+
+        """
+        return self._data.get('enabled')
+
+    @property
+    def dtstart(self):
```
Comment (costastf) [inline towerlib/entities/schedule.py:188]: please use a friendlier variable name like datetime_start

File: towerlib/entities/schedule.py [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+        """The first occurrence of the schedule occurs on or after this time.
+
+        Returns:
+            datetime: The first occurrence of the schedule occurs on or after this time.
+
+        """
+        return self._data.get('dtstart')
+
+    @property
+    def dtend(self):
```
Comment (costastf) [inline towerlib/entities/schedule.py:198]: datetime_end

File: towerlib/entities/schedule.py [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+        """Extra data to pass to the execution as a json.
+
+        Returns:
+            None.
+
+        """
+        self._update_values('enabled', value)
+
+    @rrule.setter
+    def rrule(self, value):
```
Comment (costastf) [inline towerlib/entities/schedule.py:375]: friendlier naming here

File: towerlib/entities/host.py (hunk 1 above)
Comment (SayantanKhanra10) [reply towerlib/entities/host.py:302]: Yes i just changed the logic and over-looked that it was already wrong there. Will fix.

File: towerlib/entities/schedule.py (hunk 4 above)
Comment (ilijamt) [reply towerlib/entities/schedule.py]: Sayantan already fixed this.

File: towerlib/entities/schedule.py (hunk 3 above)
Comment (ilijamt) [reply towerlib/entities/schedule.py]: @SayantanKhanra10 already fixed this.

File: towerlib/entities/schedule.py (hunk 2 above)
Comment (ilijamt) [reply towerlib/entities/schedule.py]: @SayantanKhanra10 already fixed this. But yes I do agree it's friendlier.

**Pull request merged**